### ISSUE TITLE
Devcontainer cannot start a container. Permission Denied on docker-compose.devcontainer.build.yml

### ISSUE THREAD

**Issue by Captnlink** [microsoft/vscode-remote-release#9545]
Problem is already solved, but I want to leave a trace for others to see if they find themselves in the same spot.

TLDR: Validate that you have multiple instances of docker (i.e., snap and apt), and remove one. I removed the snap version.
Use official installation guide from docker.

Problem encountered
Last week, one morning, I encountered a problem with my setup. I tried to find help in forums and GitHub issues but to no avail.

While trying to boot my environment, I am not able to create my devcontainer and log in to it as usual.
I can build and use the container just fine by itself. (Compile my app and run it inside.)

- VSCode Version:
```
Version: 1.86.1
Commit: 31c37ee8f63491495ac49e43b8544550fbae4533
Date: 2024-02-07T09:07:19.849Z
Electron: 27.2.3
ElectronBuildId: 26495564
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Linux x64 6.5.0-17-generic
```

- Local OS Version:
```
cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

- Remote OS Version:
```
ubuntu:22.04
```

- Remote Extension/Connection Type: Containers

- Docker Version:
```
docker --version
Docker version 25.0.3, build 4debf41
```

- Extension:
  - Dev Container:
```
Id: ms-vscode-remote.remote-containers
Version: 0.342.0
```
  - Dev Container:
```
Id: ms-azuretools.vscode-docker
Version: 1.28.0
```
---
- Logs:
[remoteContainers.log](https://github.com/microsoft/vscode-remote-release/files/14239794/remoteContainers.log)

Steps to Reproduce:

1. Open project [base_project.zip](https://github.com/microsoft/vscode-remote-release/files/14239774/base_project.zip)
2. ctrl+shift+p -> reopen in container
3. Error (see remoteContainers.log)

Step done to find solution include:
 - reinstall Vscode (dirty and clean version aka remove .vscode and .config)
 - reinstall docker
 - Try earlier version of extention: `dev container`
 - Try earlier version of Docker: 
 ```
 VERSION_STRING=5:24.0.0-1~ubuntu.22.04~jammy
sudo apt-get install docker-ce=$VERSION_STRING docker-ce-cli=$VERSION_STRING containerd.io docker-buildx-plugin docker-compose-plugin
 ```
- try to remove buildkit, in the build args and in the terminal

----------------
Debug Process: 
Running the following line,
```
docker-compose --project-name tests -f /home/user/Documents/tests/docker-compose.yml -f /home/user/Documents/tests/.devcontainer/docker-compose-vscode.yml -f /home/user/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data/docker-compose/docker-compose.devcontainer.build-1707753988368.yml build
```
resulted in this:
```
open /home/user/.config/Code/User/globalStorage/ms-vscode-remote.remote-containers/data/docker-compose/docker-compose.devcontainer.build-1707753988368.yml: permission denied
```

Removing `build-1707753988368.yml` file made it works. The problem seems related to permission of that particular file.

Moving the file to `/tmp` did not help.

Thinking back to the line, `docker-compose` should be `docker compose`

```
which docker-compose
/snap/bin/docker-compose

which docker
/usr/bin/docker
```
Ah! 

There are two versions of docker on this computer. It seems the latest update broke the snap version.

**Comment by bamurtaugh** [comment]
Thank you for taking the time to share these steps. If you think there are any tips/info missing from our docs, we'd absolutely appreciate a PR (and I'm happy to review): https://github.com/microsoft/vscode-docs.

Closing for now, but please let us know too if you're still experiencing any issues.
